src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,287 @@
+#modes = cluster.Enums.OperatingModeEnum
+#features = cluster.Bitmaps.Feature
+
+class TC_BRBINFO_4_1(MatterBaseTest):
+
+    #
+    # Class Helper functions
+    #
+
+    async def _read_attribute_expect_success(self, endpoint, cluster, attribute, node_id):
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute, node_id=node_id)
+
+    #
+    # Test Harness Helpers
+    #
+
+    def desc_TC_BRBINFO_4_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC_BRBINFO_4_1] Verification of KeepActive Command [DUT-Server]"
+
+    def steps_TC_BRBINFO_4_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning of the DUT, already done", is_commissioning=True),
+            TestStep(2, "Commissioning of the ICD, already done", is_commissioning=True),
+            TestStep(3, "TH reads from the ICD the IDLE_MODE_DURATION and ACTIVE_MODE_DURATION attributes"),
+            TestStep(4, "Simple KeepActive command w/ subscription. ActiveChanged event received by TH contains PromisedActiveDuration"),
+            TestStep(5, "Multiple KeepActive commands. ActiveChanged event received by TH contains a PromisedActiveDuration"),
+            TestStep(6, "KeepActive not returned after 60 minutes of offline ICD"),
+        ]
+        return steps
+
+
+    def pics_TC_BRBINFO_4_1(self) -> list[str]:
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        pics = [
+            "BRBINFO.C.80",
+        ]
+        return pics
+
+    def _ask_for_vendor_commissioniong_ux_operation(self, discriminator, setupPinCode):
+        self.wait_for_user_input(
+            prompt_msg=f"Using the DUT vendor's provided interface, commission the ICD device using the following parameters:\n"
+                        f"- discriminator: {discriminator}\n"
+                        f"- setupPinCode: {setupPinCode}\n"
+                        f"If using FabricSync Admin, you may type:\n"
+                        f">>> pairing onnetwork 111 {setupPinCode}")
+
+    def _display_message_and_wait_for_user_input(self, message):
+        self.wait_for_user_input(prompt_msg=message)
+
+    async def _send_keep_active_command(self, duration):
+        logging.info("Mock sending keep active command")
+        # TODO check success
+        #keepActive = await self.default_controller.SendCommand(nodeid=self.dut_node_id, endpoint=kICDBridgedEndpointId, payload=brbinfoCluster.Commands.KeepActive(stayActiveDuration=duration))
+
+    async def _wait_for_active_changed_event(self, timeout) -> int:
+        try:
+            PromisedActiveDuration = self.q.get(block=True, timeout=timeout)
+            logging.info(f"PromisedActiveDuration: {PromisedActiveDuration}")
+            return PromisedActiveDuration
+        except queue.Empty:
+            asserts.assert_true(False, "Timeout on event")
```

Comment:
nit, you can assert_fail directly
```suggestion
            asserts.assert_fail("Timeout on event")
```